projects/ngrx.io/content/guide/component-store/lifecycle.md

```diff
@@ -0,0 +1,153 @@
+# Lifecycle
+
+NgRx ComponentStore comes with lifecycle hooks and observables for performing tasks after the ComponentStore is initially instantiated, after the initial state is first set, and when the ComponentStore is destroyed. You can use these lifecycle hooks to set up long-running effects, wire up additional logic, and other tasks outside the constructor of the ComponentStore.
+
+## Setup
+
+Both lifecycle hooks are enabled by providing the ComponentStore through the `provideComponentStore()` function. This function registers the ComponentStore as a provider, and sets up a factory provider to instantiate the ComponentStore instance and run the implemented lifecycle hooks.
+
+Currently, Angular provides initializer tokens in a few areas. The `APP_INITIALIZER` and `BOOTSTRAP_INITIALIZER` for application/bootstrap init logic, and the `ENVIRONMENT_INITIALIZER` for environment injector init logic. The `provideComponentStore()` mimics this behavior to run the lifecycle hooks. The function is required because there aren't any provided tokens at the component level injector to allow initialization tasks.
+
+<div class="alert is-important">
+
+**Note:** If you implement the lifecycle hooks in the ComponentStore, and register it with `providers` without using `provideComponentStore()`, in development mode, a warning is logged to the browser console.
+
+</div>
+
+## OnStoreInit
+
+The `OnStoreInit` interface is used the implement the `ngrxOnStoreInit` method in the ComponentStore class. This lifecycle method is called immediately after the ComponentStore class is instantiated.
+
+```ts
```

Comment:
Should we use the `<code-example>` tag instead to keep things consistant (and split the examples in two)?

For comparison:
![image](https://user-images.githubusercontent.com/28659384/170764164-8b8e9be2-38e1-426f-bc64-b4f0696db8ca.png)